### DISCUSSION TITLE
Mobile - Offline first - Sync remote Postgres DB to a local SQLite DB

### DISCUSSION THREAD

**Discussion by mohamad-jawad** [supabase/supabase#14102] [Questions]
Hello Supabase community, I am currently working on a Flutter mobile project that supports offline-first.

On every opening of the app (or refresh button tapped), I am sending a read query for each table one after the other, getting the last updated rows depending on `last_update_time` that I can get from my local tables separately.

So each response of the read query will contains the last updated rows of the corresponding table. after that I update my local DB with these rows.

The question is:
is it better and faster If I create an edge function that takes as parameters `last_update_time`  of all the local tables, and this edge function will call the read query of all the remote tables then it returns to the app all the updated rows of all the tables at once ?

Or should I create an RPC function instead of the edge function ?

Or is the current approach is enough ?

Thanks in advance for any suggestions.

**Comment by Auto-AGI** [comment]
Have you considered using a synchronization library, such as **SyncAdapter** or **Firebase Sync**, to handle the synchronization process? What is the expected size of the data that needs to be synchronized, and how often will the synchronization process occur?
